src/Pirouette/Term/ConstraintTree.hs

```diff
@@ -150,14 +150,15 @@ pruneMaybe (Choose x ty ts) =
   where choose [] = fail "empty tree"
         choose ts = return (Choose x ty ts)
 
-termToCTree :: (MonadPirouette m) => CTreeOpts -> Name -> PrtTerm -> m (CTree Name)
-termToCTree opts name t = do
-  let (args, body) = R.getHeadLams (R.appN t $ map (R.Arg . flip R.App [] . flip R.B 0 . R.Ann . fromString) (coWithArguments opts))
-  unless (null args) $
-    logWarn $ "Executing a non-saturated term. Will use the variable names: " ++ show args
-  mctree <-
-    execute body >>= runMaybeT . if coPruneMaybe opts then pruneMaybe else return
-  logDebug $ "Translating Constraint Tree for " ++ show name
-  case mctree of
-    Nothing -> throwError' $ PEOther "termToCTree: empty tree"
-    Just tr -> return tr
+termToCTree :: (MonadPirouette m) => CTreeOpts -> Name -> PrtDef -> m (CTree Name)
```

Comment:
nipicking it at this point, but if the function is `termToCTree`, I'd expect to receive a `PrtTerm`, not a `PrtDef`. :thinking:

Reply:
Agree with you that the name should maybe be changed, but since it uses both the definition body and its type, it seems more logical to me to pass it a `PrtDef` rather than at each of the 2 use of this function do a 
```
case def of 
  DFunction _ t ty -> termToCTree opts n t ty
  _ -> throwError' "expected a function"
```